### PR TITLE
TDL-19292 Add new cards stream and removed deprecated streams.

### PR DESCRIPTION
**Pull request by prijendev**
# Description of change
**TDL-19291 Remove deprecated streams and fields** 
- Removed the following deprecated streams,
  -  `Website Cards `
  - `Video Website Cards `
  - `Image App Download Cards `
  - `Video App Download Cards `
  - `Card image direct message`
  - `Video direct message`
- Removed following fields from campaign stream,
  - `start_time`
  - `end_time`

**TDL-19292 Add new field and stream**
- Added [cards](https://developer.twitter.com/en/docs/twitter-ads-api/creatives/api-reference/cards#cards) stream and it's schema.
- Added the `budget_optimization` field into the schema of the campaign stream. 
- Passed all integration test cases in the local environment, that are added in [PR#20](https://github.com/singer-io/tap-twitter-ads/pull/20).

# Manual QA steps
 - Run the discover mode and verify that removed streams are not available in the catalog and the `cards` stream is available in the catalog.
 - Run the sync mode and verify that tap write all records of the `cards` stream. Also, verify that the `cards` stream respects the bookmark key.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_twitter_ads/schemas/cards.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            "type": "array",
+            "items": {
+              "type": ["null", "object"],
+              "properties": {}
+            }
+          },
+          {
+            "type": "null"
+          }
+        ]
```
Comment (namrata270998) [inline tap_twitter_ads/schemas/cards.json:20]: We don't need to define `null` separately inside `anyof`, instead we can just do:
`"type": ["null", "array"],
"items": {
              "type": ["null", "object"],
              "properties": {}
            }`

Comment (namrata270998) [inline tap_twitter_ads/schemas/cards.json:20]: We don't need to define `null` separately inside `anyof`, instead we can just do:
`"type": ["null", "array"],
"items": {
              "type": ["null", "object"],
              "properties": {}
            }`

Comment (namrata270998) [inline tap_twitter_ads/schemas/cards.json:20]: We don't need to define `null` separately inside `anyof`, instead we can just do:
```

"type": ["null", "array"],
"items": {
              "type": ["null", "object"],
              "properties": {}
            }

Comment (namrata270998) [inline tap_twitter_ads/schemas/cards.json:20]: We don't need to define `null` separately inside `anyof`, instead we can just do:
```

"type": ["null", "array"],
"items": {
              "type": ["null", "object"],
              "properties": {}
            }

**Review by namrata270998 — CHANGES_REQUESTED** [review]
We need to change the schema for array types

File: tap_twitter_ads/schemas/cards.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "additionalProperties": false,
+    "properties": {
+      "name": {
+        "type": ["null", "string"]
+      },
+      "components": {
+        "type": ["null", "array"],
+        "items": {
+          "type": ["null", "object"],
+          "properties": {}
```
Comment (savan-chovatiya) [inline tap_twitter_ads/schemas/cards.json:12]: Why did we keep empty properties here?

File: tap_twitter_ads/schemas/line_items.json [hunk 3]
```diff
@@ -8,9 +8,6 @@
     "bid_type": {
       "type": ["null", "string"]
     },
-    "bid_strategy": {
-      "type": ["null", "string"]
-    },
```
Comment (savan-chovatiya) [inline tap_twitter_ads/schemas/line_items.json:13]: Why did we remove this field from the line_items? The line_items is not used in cards.

**Review by savan-chovatiya — CHANGES_REQUESTED** [review]

File: tap_twitter_ads/schemas/campaigns.json [hunk 4]
```diff
@@ -8,10 +8,6 @@
     "name": {
       "type": ["null", "string"]
     },
-    "start_time": {
```
Comment (dbshah1212) [inline tap_twitter_ads/schemas/campaigns.json:11]: Any reason of removing these fields?

File: tap_twitter_ads/schemas/cards.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (prijendev) [reply tap_twitter_ads/schemas/cards.json]: Updated.

**Comment by prijendev** [comment]
> We need to change the schema for array types

Changed schema for array types.

File: tap_twitter_ads/schemas/line_items.json [hunk 5]
```diff
@@ -8,9 +8,6 @@
     "bid_type": {
       "type": ["null", "string"]
     },
-    "bid_strategy": {
```
[the remark this answers is not in the document]
Comment (prijendev) [reply tap_twitter_ads/schemas/line_items.json]: Added back the `bid_strategy` field.

File: tap_twitter_ads/schemas/campaigns.json (hunk 4 above)
Comment (prijendev) [reply tap_twitter_ads/schemas/campaigns.json:11]: `start_time` and `end_time` fields are deprecated. ([link](https://twittercommunity.com/t/ads-api-version-11/168814#changed-moving-start-end-times-to-line-item-level-6)) That's why removed it from the schema.

File: tap_twitter_ads/schemas/line_items.json (hunk 3 above)
Comment (prijendev) [reply tap_twitter_ads/schemas/line_items.json]: Added back the `bid_strategy` field.

File: tap_twitter_ads/schemas/cards.json (hunk 2 above)
Comment (prijendev) [reply tap_twitter_ads/schemas/cards.json:12]: There is no specific document available for the schema of the `cards` stream. In the SDK, they mentioned just the `components` field with the `list` datatype. That's why we kept empty properties here.

File: tap_twitter_ads/schemas/campaigns.json (hunk 4 above)
Comment (dbshah1212) [reply tap_twitter_ads/schemas/campaigns.json:11]: Got it, make sure you added those fields inside Line Item.

**Review by dbshah1212 — APPROVED** [review]

**Review by kspeer825 — APPROVED** [review]

**Review by RushT007 — APPROVED** [review]

**Review by KrisPersonal — APPROVED** [review]

**Pull request merged**